src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs

```diff
@@ -280,8 +281,9 @@ public async Task<IActionResult> Delete(string menuContentItemId, string menuIte
                 return NotFound();
             }
 
-            var menuItems = menuContentAsJson[nameof(MenuItemsListPart)]["MenuItems"] as JsonArray;
-            menuItems.Remove(menuItem);
+            var menuItemsListPart = menuContentAsJson[nameof(MenuItemsListPart)] as JsonDynamicObject;
```

Comment:
Please keep the original code
We add Remove and other functions just to keep untransformed code like `menu.Content.Remove` working